exasol_advanced_analytics_framework/udf_communication/background_listener.py

```diff
@@ -0,0 +1,110 @@
+import threading
+import traceback
+from typing import Optional, Iterator
+
+import structlog
+import zmq
+from structlog.types import FilteringBoundLogger
+
+from exasol_advanced_analytics_framework.udf_communication.background_listener_run import BackgroundListenerRun
+from exasol_advanced_analytics_framework.udf_communication.connection_info import ConnectionInfo
+from exasol_advanced_analytics_framework.udf_communication.ip_address import IPAddress
+from exasol_advanced_analytics_framework.udf_communication.messages import Message, StopMessage, RegisterPeerMessage, \
+    MyConnectionInfoMessage
+from exasol_advanced_analytics_framework.udf_communication.peer import Peer
+from exasol_advanced_analytics_framework.udf_communication.serialization import deserialize_message, serialize_message
+
+LOGGER: FilteringBoundLogger = structlog.get_logger()
+
+
+class BackgroundListener:
```

Comment:
Actually, I am thinking, maybe it would be good to rename that class to something more specific, maybe "PeerCommunicationBackgroundListener".

Reply:
moved PeerCommunicator and BackgroundListener to the peer_communicator module, because the names got otherwise to long